service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java

```diff
@@ -67,184 +67,184 @@ static void setUp() {
+            when(tagRepository.findByCertificateId(1L)).thenReturn(createTagList());
 
-        when(certificateValidator.isCertificateDtoValid(certificateDto)).thenReturn(true);
+            CertificateDto actualCertificateDto = certificateService.findCertificateById(certificateId);
 
-        Assertions.assertThrows(EntityAlreadyExistsException.class, () -> certificateService.create(certificateDto));
-    }
+            verify(certificateRepository, times(expectedInteractions)).findById(1L);
+            verify(tagRepository, times(expectedInteractions)).findByCertificateId(1L);
 
-    @Test
-    void updateCertificateTest() {
-        long certificateId = 0;
-        Certificate certificate = createCertificate();
-        CertificateDto updatedCertificateDto = createCertificateDto();
-        updatedCertificateDto.setId(certificateId);
+            Assertions.assertEquals(expectedCertificateDto.toString(), actualCertificateDto.toString());
+        }
 
-        when(certificateRepository.update(certificate)).thenReturn(true);
-        when(certificateValidator.isCertificateDtoValid(updatedCertificateDto)).thenReturn(true);
-        assertTrue(certificateService.update(updatedCertificateDto));
-    }
+        @Test
+        void findByIdWhenCertificateNotFoundTest() {
+            int certificateId = 1;
+            when(certificateRepository.findById(anyLong())).thenReturn(Optional.empty());
+
+            Assertions.assertThrows(EntityNotFoundException.class, () -> certificateService.findCertificateById(certificateId));
+        }
+
+        @Test
+        void createCertificateTest() {
+            CertificateDto certificateDto = createCertificateDto();
+
+            when(certificateValidator.isCertificateDtoValid(certificateDto)).thenReturn(true);
 
-    @Test
-    void updateInvalidCertificateTest() {
-        CertificateDto updatedCertificateDto = createCertificateDto();
-        updatedCertificateDto.setName("");
+            certificateService.create(certificateDto);
+
+            verify(certificateRepository, times(expectedInteractions)).create(any());
+        }
 
-        Assertions.assertThrows(InvalidEntityException.class, () -> certificateService.update(updatedCertificateDto));
-    }
+        @Test
+        void createInvalidCertificateTest() {
+            CertificateDto certificateDto = null;
 
-    @Test
-    void updateInvalidTagTest() {
-        List<String> stringList = new ArrayList<>();
-        CertificateDto updatedCertificateDto = createCertificateDto();
-        stringList.add("");
-        updatedCertificateDto.setTagList(stringList);
+            Assertions.assertThrows(InvalidEntityException.class, () -> certificateService.create(certificateDto));
+        }
 
-        Assertions.assertThrows(InvalidEntityException.class, () -> certificateService.update(updatedCertificateDto));
-    }
+        @Test
+        void createAlreadyExistsCertificateTest() {
```

Comment:
https://stackoverflow.com/questions/155436/unit-test-naming-best-practices
Edit the names of your test methods according to the accepted norms